src/Stryker.Core/Stryker.Core/MutantFilters/DiffMutantFilter.cs

```diff
@@ -86,7 +86,7 @@ public IEnumerable<Mutant> FilterMutants(IEnumerable<Mutant> mutants, ReadOnlyFi
 
         private IEnumerable<Mutant> SetNotRunMutantsToIgnored(IEnumerable<Mutant> mutants)
         {
-            foreach (var mutant in mutants.Where(m => m.ResultStatus == MutantStatus.NotRun))
+            foreach (var mutant in mutants.Where(m => m.ResultStatus == MutantStatus.NotRun || m.ResultStatus == MutantStatus.NoCoverage))
```

Comment:
I think this breaks the baseline diff feature as now any nocoverage results from previous runs will be set to ignored. While that is semi-correct it is not fully correct. @Stijn-Rutten

Reply:
My understanding is that the purpose of this function is to mark mutations that are not changed. 
At this moment in Stryker test cycle, mutants are marked as not run, except if there is no coverage, in which case they have already been flagged as 'NoCoverage'.

A nicer approach may be to delay the 'NoCoverage' flagging to the test phase, but filtering them out earlier helps keep the test phase clean; and implies a significant refactoring effort

Reply:
Mhm I see your point and also just now realize that I myself said it was a bug before. However we do need to take a close look at how this interacts with the baseline diff feature and I'm also realizing that some of the other helpers in this class might be wrong. Such as SetMutantStatusForNonCSharpFileChanged, which currently overwrites the calculated NoCoverage status which is not correct I think.

Reply:
It turns out I am not familiar with the diff workflows as I always test everything. Hence I used unit test to fix this, I think it would be faster if someone more familiar with the feature took his PR or the fix over.

I can still look into this, but I am busy with other stuff right now, so not sure when I will have a proper testing setup for this.

Reply:
I think the pr is good as it is but I'll take a close look when I review 👌

Reply:
I have been testing this solution and this causes NoCoverage results to be set to Ignored in all cases. This does not change the mutation score but it does mean that if all your new code is uncovered you will not have any result on your new code. Everything will be marked as Ignored due to no change.

Reply:
so, if I understand you correctly, it sadly does not correct anything but it makes things a bit worse
am I right? 🤣

Reply:
Well it fixes the reported problem but causes a problem for the other causes so I guess net change is 0 :)

Reply:
this is odd: I assume this function was used to flag as ignored mutants that belonged to non changed files.
it should change the status of other mutations.

What am I missing here?

Reply:
You are correct, I am investigating

Reply:
The problem was caused because other helper functions did not ignore NoCoverage results. The result is now correct.

My first comment stands though, this breaks NoCoverage results when combined with the Dashboard Compare feature. I think that feature is less important so I'll create a new bug report for it.